R/tm_g_bivariate.R

```diff
@@ -363,7 +381,52 @@ srv_g_bivariate <- function(id,
       color = color, fill = fill, size = size
     )
 
-    selector_list <- teal.transform::data_extract_multiple_srv(data_extract, data)
+    rule_var <- function(other) {
+      function(value) {
+        othervalue <- selector_list()[[other]]()$select
+        if (length(value) == 0L && length(othervalue) == 0L)
+          "Please select at least one of x-variable or y-variable"
+      }
+    }
+    rule_diff <- function(other) {
+      function(value) {
+        if (!is.null(selector_list()[[other]]()$select)) {
+          othervalue <- selector_list()[[other]]()$select
+          if (identical(value, othervalue))
+            "Row and column facetting variables must be different."
+        }
+      }
+    }
+
+    selector_list <- teal.transform::data_extract_multiple_srv(
+      data_extract = data_extract,
+      datasets = data,
+      select_validation_rule = list(
+        x = rule_var("y"),
+        y = rule_var("x"),
+        row_facet = shinyvalidate::compose_rules(
+          shinyvalidate::sv_optional(),
+          ~ if (length(.) > 1) "There must be 1 or no row facetting variable.",
```

Comment:
Here and line 414 much better to remove rule here and to check [here](https://github.com/insightsengineering/teal.modules.general/blob/main/R/tm_g_bivariate.R#L117-L137) like we do for all the other d-e-s in this module?